### PR TITLE
DISPLAY-990: Fixed unknown class in data providers

### PR DESCRIPTION
**Pull request by cableman**
#### Link to ticket

https://jira.itkdev.dk/browse/DISPLAY-990

#### Description

Code clean up in feeds handling.

#### Screenshot of the result

N/A

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

N/A

### CODE REVIEW

File: src/Service/FeedService.php [hunk 1]
```diff
@@ -62,50 +75,113 @@ public function getRemoteFeedUrl(Feed $feed): string
...
                         $cacheItem->expiresAfter($feedConfiguration['cache_expire']);
                     }
-
                     $this->feedsCache->save($cacheItem);
 
                     return $data;
                 }
             }
 
+            // If feed type was not known in the system return null. API platform will convert this to 404 not found.
```
Comment (jekuaitk) [inline src/Service/FeedService.php:144]: 👍

**Review by jekuaitk — APPROVED** [review]
A couple comments/considerations, but looks good to me.

File: src/Feed/KobaFeedType.php [hunk 2]
```diff
@@ -26,48 +27,53 @@ public function __construct(
...
-        $secrets = $feedSource->getSecrets();
+        $secrets = $feedSource?->getSecrets();
         $configuration = $feed->getConfiguration();
 
         if (!isset($secrets['kobaHost']) || !isset($secrets['kobaApiKey'])) {
-            $this->logger->error('KobaFeedType: Feed source not configured.');
+            $this->logger->error('KobaFeedType: kobaHost and kobaApiKe not configured.');
 
-            return [];
+            throw new MissingFeedConfiguration('kobaHost and kobaApiKe not configured');
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Feed/KobaFeedType.php]: Fixed

File: src/Feed/KobaFeedType.php [hunk 3]
```diff
@@ -26,48 +27,53 @@ public function __construct(
...
+        $results = [];
+
         $feedSource = $feed->getFeedSource();
-        $secrets = $feedSource->getSecrets();
+        $secrets = $feedSource?->getSecrets();
         $configuration = $feed->getConfiguration();
 
         if (!isset($secrets['kobaHost']) || !isset($secrets['kobaApiKey'])) {
-            $this->logger->error('KobaFeedType: Feed source not configured.');
+            $this->logger->error('KobaFeedType: kobaHost and kobaApiKe not configured.');
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Feed/KobaFeedType.php]: Fixed

File: src/Exceptions/MissingFeedConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace App\Exceptions;
+
+class MissingFeedConfiguration extends \Exception
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Exceptions/MissingFeedConfiguration.php]: Fixed

File: src/Exceptions/UnknownFeedType.php [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace App\Exceptions;
+
+class UnknownFeedType extends \Exception
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Exceptions/UnknownFeedType.php]: Fixed

**Pull request merged**